src/main/java/de/flux/playground/deckcompare/fileupload/FileUploadController.java

```diff
@@ -0,0 +1,67 @@
+package de.flux.playground.deckcompare.fileupload;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.stream.Collectors;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.util.FileCopyUtils;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import de.flux.playground.deckcompare.Deckcompare;
+
+@Controller
+public class FileUploadController {
+
+    @RequestMapping(method = RequestMethod.GET, value = "/upload")
+    public String provideUploadInfo(Model model) {
+        File rootFolder = new File(Deckcompare.ROOT);
+
+        model.addAttribute("files",
+                           Arrays.stream(rootFolder.listFiles())
+                                 .sorted(Comparator.comparingLong(f -> -1 * f.lastModified())).map(f -> f.getName())
+                                 .collect(Collectors.toList()));
+
+        return "uploadForm";
+    }
+
+    @RequestMapping(method = RequestMethod.POST, value = "/upload")
+    public String handleFileUpload(@RequestParam("name") String name, @RequestParam("file") MultipartFile file,
+                                   RedirectAttributes redirectAttributes) {
+        if (name.contains("/")) {
+            redirectAttributes.addFlashAttribute("message", "Folder separators not allowed");
+            return "redirect:upload";
+        }
+        if (name.contains("/")) {
```

Comment:
This is a copy and paste bug. The code is from https://spring.io/guides/gs/uploading-files/